setup.py

```diff
@@ -57,6 +55,81 @@ def dmf_data_files(root: str = DMF_DATA_ROOT) -> List[Tuple[str, List[str]]]:
     return file_list
 
 
+class ExtraDependencies:
+    """
+    A convenience shorthand to define and combine dependencies for ``extras_require``.
+
+    >>> extras = ExtraDependencies()
+    >>> extras["ui"]
+    ['requests', 'pint']
+    >>> list(extras)
+    ['ui', 'examples', 'dmf', 'surrogate', 'mlai', 'grid', 'optional']
+    >>> dict(extras)
+    {'ui': ['requests', 'pint'], 'examples': ['jupyter', 'nbconvert', 'nbformat'], ...}
+    """
+
+    ui = [
+        "requests",
+        "pint",
```

Comment:
This might fall out else where, but `pint` is something we use fairly broadly (if maybe indirectly through Pyomo). I do believe that we use `pint` directly in a few places.

Reply:
> This might fall out else where, but `pint` is something we use fairly broadly (if maybe indirectly through Pyomo). I do believe that we use `pint` directly in a few places.

That's a good point. `pint` is already part of the default dependencies so having it repeated here is not strictly needed. The reason why I added is an attempt to be explicit, i.e. "if you're importing it, it should be in your dependencies", and `pint` is imported and used directly in the `idaes.core.ui` code.

Note that this is anyway somewhat of a moot point, as this dependency will become `idaes-ui` once we complete the migration of the UI codebase to IDAES/idaes-ui (#1146).